src/patients/hooks/useAddCareGoal.tsx

```diff
@@ -0,0 +1,47 @@
+import { isEmpty } from 'lodash'
+import { queryCache, useMutation } from 'react-query'
+
+import PatientRepository from '../../shared/db/PatientRepository'
+import CareGoal from '../../shared/model/CareGoal'
+import { uuid } from '../../shared/util/uuid'
+import validateCareGoal from '../util/validate-caregoal'
+
+interface AddCareGoalRequest {
+  patientId: string
+  careGoal: Omit<CareGoal, 'id' | 'createdOn'>
+}
+
+async function addCareGoal(request: AddCareGoalRequest): Promise<CareGoal[]> {
```

Comment:
there should be tests for this hook

Reply:
Yeah, I'd forgotten and then I did it on the lastest commit.